src/redux/reducers/extensions.js

```diff
@@ -1,18 +1,25 @@
-const defaultState = {};
+export const defaultState = {}
 
 const extensions = (state = defaultState, action) => {
   switch (action.type) {
     case 'SET_EXTENSION_DATA':
-      const extensionName = action.extension;
-      const currentData = state[extensionName] || {};
-      const updatedData = Object.assign({}, currentData, action.data);
-      const updatedState = state;
-      state[extensionName] = updatedData;
+      const extensionName = action.extension
+      const currentData = state[extensionName] || {}
+
+      const incomingData = action.data
+
+      const extension = {
+        [extensionName]: {
+          ...currentData,
+          ...incomingData,
+        },
+      }
+
+      return { ...state, ...extension }
```

Comment:
It's always nice to see code written in different ways. The `...` does feel much cleaner ^_^